aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -232,6 +232,26 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+void cummax_helper_cpu(const Tensor& self, Tensor& values, Tensor& indices, int64_t dim) {
+  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Bool, at::ScalarType::Half,
+    self.scalar_type(), "cummax_cpu", [&]() -> void {
+    at::native::tensor_dim_apply3<scalar_t, scalar_t>(self, values, indices, dim,
+      [](const scalar_t* self_data, scalar_t* values_data, int64_t* indices_data,
+        int self_dim_size, int self_stride, int values_stride, int indices_stride) {
+        auto cummax = self_data[0];
+        int idx = 0;
+        for(int i = 0; i < self_dim_size; i++) {
+          if(self_data[i*self_stride] >= cummax) {
```

Comment:
you have to correctly handle `nan` values - they should always win. With this comparison, `nan` will essentially be ignored